### ISSUE TITLE
Standard deployment failures

### ISSUE THREAD

**Issue by luticm** [AzureCosmosDB/ClaimsProcessing#30]
Opening item for standard deployment failures.
I'm not including other errors as they can be a consequence of this one.

Cloud shell and local deployments are failing with same error:

**Cloud Shell**
ERROR: {"code": "InvalidTemplateDeployment", "message": "The template deployment 'main' is not valid according to the validation procedure. The tracking id is 'c6653d48-350d-4e8e-846a-37ae375adf36'. See inner errors for details."}

Inner Errors: 
{"code": "ApiSetDisabledForCreation", "message": "It's not allowed to create new accounts with type 'OpenAI'."}
Retrieving Aks Name
The name of your AKS: 
argument --name/-n: expected one argument

**Comment by joelhulen** [comment]
@luticm when you cloned the repo, did you include the `--recurse-submodules` parameter?

Example:
`git clone --recurse-submodules https://github.com/AzureCosmosDB/ClaimsProcessing.git`

Also, do you have the capacity in your target Azure subscription to deploy more Azure OpenAI accounts?

**Comment by luticm** [comment]
Yes, repo was cloned with submodules. OpenAI capacity and model restrictions are challenging to deal during deployments.
I was able to successful deploy OpenAI on EastUS.

The original setup script allowed to specify existing OpenAI account by using parameters: -openAiRg <rg> -openAiName <name> and -openAiDeployment 'completions'.
Bringing back those options can definitely facilitate deployment, due to existing restrictions.

@joelhulen - _Can we add this option to this repo and others reference apps?_

**Comment by luticm** [comment]
All synapse dataset deployment fails due to path.

"Failed to parse string as JSON:
@./dataset/C:\Arquivos\Projects\Git\Microsoft\ClaimsProcessing\synapse\dataset\AdlsAdjudicator.json"

@fonsecamar suggested a fix for this a while back. Just placed a PR with suggestion.

Let me know if you prefer to track this under a new issue.

**Comment by joelhulen** [comment]
> Yes, repo was cloned with submodules. OpenAI capacity and model restrictions are challenging to deal during deployments. I was able to successful deploy OpenAI on EastUS.
> 
> The original setup script allowed to specify existing OpenAI account by using parameters: -openAiRg -openAiName and -openAiDeployment 'completions'. Bringing back those options can definitely facilitate deployment, due to existing restrictions.
> 
> @joelhulen - _Can we add this option to this repo and others reference apps?_

We removed the ability to specify an existing OpenAI account. This is because that process was very brittle and randomly failed vs. always deploying an OpenAI account. I would also like to see that capability return, but might need to work on that in the near future.

**Comment by sesmyrnov** [comment]
@joelhulen  - can you merge/add the fix for powershell ""Failed to parse string as JSON:" which Luciano mentioned earlier we had a fix submitted as PR in the orig repos? That way we can at least eliminate this set of errors and focus on what is left?

**Comment by joelhulen** [comment]
@sesmyrnov I did merge in his PowerShell fix. However, when I ran a deployment after the path change, it was unable to find and parse any of the JSON files. I reversed the change (https://github.com/AzureCosmosDB/ClaimsProcessing/pull/42), which fixed the Synapse deployment.
